internal/monitor/monitor.go

```diff
@@ -0,0 +1,110 @@
+package monitor
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/castai/cluster-controller/internal/config"
+)
+
+func Run(ctx context.Context, log logrus.FieldLogger, clientset *kubernetes.Clientset, metadataFile string, pod config.Pod, clusterIDHandler func(clusterID string)) error {
+	m := monitor{
+		clientset: clientset,
+		log:       log,
+		pod:       pod,
+	}
+
+	metadataUpdates, err := watchForMetadataChanges(ctx, m.log, metadataFile)
+	if err != nil {
+		return fmt.Errorf("setting up metadata watch: %w", err)
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case metadata := <-metadataUpdates:
+			clusterIDHandler(metadata.ClusterID)
```

Comment:
Why do we need to update clusterID base on metadata file changes. ClusterID should be passed via env on the start and shouldn't change

Reply:
Let me clean this logic, which left after the agent